tools/GenerateCompilationDatabase.ps1

```diff
@@ -0,0 +1,166 @@
+Param(
+    [string]$Ms2ccVersion = "1.3.0",
+    [switch]$Update
+
+)
+
+$binlogFileBase = Join-Path (Split-Path $PSScriptRoot -parent) "BuildOutput\Binlogs"
+$binlogFiles = Get-ChildItem -Path $binlogFileBase -Filter *.binlog -Recurse | Select-Object -ExpandProperty FullName
+
+Write-Host "Binlog files found:"
+$binlogFiles | ForEach-Object { Write-Host $_ }
+
+$VCToolsInstallDir = . "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe" -Latest -prerelease -requires Microsoft.Component.MSBuild -property InstallationPath
+write-host "VCToolsInstallDir: $VCToolsInstallDir"
+
+$msBuildPath = "$VCToolsInstallDir\MSBuild\Current\Bin\msbuild.exe"
+write-host "msBuildPath: $msBuildPath"
+
+Remove-Item "temp-filtered.log" -ErrorAction SilentlyContinue
+Remove-Item "temp-filtered2.log" -ErrorAction SilentlyContinue
+
+# Target "ClCompile" in file "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets" from project "C:\WindowsAppSDK\dev\Detours\Detours.vcxproj" (target "_ClCompile" depends on it):
+# Context here would be: "C:\WindowsAppSDK\dev\Detours"
+
+foreach ($binlogFile in $binlogFiles) {
+    if (-Not (Test-Path $binlogFile)) {
+        Write-Error "Binlog file not found: $binlogFile"
+        exit 1
+    }
+
+    & $msBuildPath $binlogFile /v:normal /noconlog /flp:logfile=temp.log
+
+    Select-String -Path "temp.log" -Pattern "Target ""ClCompile"" in file","Cl.exe"  |
+        ForEach-Object { $_.Line } |
+        Where-Object { $_ -notmatch "Tracker.exe" } |
+        Out-File -FilePath "temp-filtered.log" -Append -Encoding utf8
+
+    Remove-Item "temp.log"
+
+    Write-Host "Processed binlog file: $binlogFile"
+}
+
+Write-Host "Filtered log file generated at: $(Get-Location)\temp-filtered.log"
+
+$contextPath = ""
+
+# for each line in temp-filtered.log, if it is a "Target " line, extract the project directory and save in the contextPath variable
+# if it is a "Cl.exe" line, prepend the contextPath to every source file in that line only if it is
+# a relative path, then output the modified line to temp-filtered-2.log
+# The source files are the arguments that ends with .cpp, .c, or .h
+# They can have spaces in them. If so, they are enclosed in quotes.
+
+$lines = Get-Content "temp-filtered.log"
+
+Write-Host "Processing filtered log to adjust file paths..."
+
+foreach ($line in $lines) {
+    if ($line -match 'Target "ClCompile" in file "([^"]+)" from project "([^"]+)"') {
+        $contextPath = Split-Path $matches[2] -parent
+        Write-Host "Context path set to: $contextPath"
+    } elseif ($line -match 'Cl\.exe (.+)$') {
+        $clLine = $matches[1]
+        $args = $clLine -split ' (?=(?:[^"]*"[^"]*")*[^"]*$)'
+
+        $modifiedArgs = @()
+        foreach ($arg in $args) {
+            if ($arg -match '^(.*\.(cpp|c|h))$' -or $arg -match '^"(.*\.(cpp|c|h))"$') {
```

Comment:
no `.hpp`?

(`.cxx` and `.hxx` are older school, winappsdk has none and we're unlikely to at this point in the industry)